cluster/saltbase/salt/kube-addons/init.sls

```diff
@@ -20,6 +20,18 @@
     - file_mode: 644
 {% endif %}
 
+{% if pillar.get('enable_cluster_monitoring', '').lower() == 'googleinfluxdb' %}
+/etc/kubernetes/addons/cluster-monitoring/googleinfluxdb:
+  file.recurse:
+    - source: salt://kube-addons/cluster-monitoring
```

Comment:
In order to enable both GCM/GCL and InfluxDB/Grafana we need to enable all the services/controllers for both (meaning we need everything in `cluster/addons/cluster-monitoring/influxdb` and `cluster/addons/cluster-monitoring/google`) *except* for the `heapster-controller` files which need to be modified to be able to pass to both backends instead of just one or the other (hence the new `heapster-controller-combined` file).

So when `googleinfluxdb` is specified, we pick up all the files under `cluster/addons/cluster-monitoring/*` including the `google`, `influxdb`, and the new `googleinfluxdb` folders, but we exclude the `heapster-controller` files (`googleinfluxdb` contains the new combined `heapster-controller-combined` file).

An alternative would be to copy all the non-`heapster-controller` files from influxdb to the new `googleinfluxdb` folder. That way we could just include `googleinfluxdb` folder and be done. But I'd prefer not to do that (people will have to remember to modify both instances of each file). Let me know if you feel strongly otherwise.